src/services/identity/UsersService.ts

```diff
@@ -155,26 +155,46 @@ class UsersService {
 
   async canSendEmailOtp(email: string) {
     const normalizedEmail = email.toLowerCase()
-    const whitelistEntries = await this.whitelist.findAll({
-      attributes: ["email"],
-      where: {
-        expiry: {
-          [Op.or]: [{ [Op.is]: null }, { [Op.gt]: new Date() }],
+
+    // raw query because ORMs suck!
```

Comment:
Just curious we used raw query here because the ORM way was more complicated or?

Reply:
Well, so this query has 2 query patterns which are "unusual" for typical ORM use-cases:
1. a `CASE WHEN` in the `WHERE` clause
2. a where condition of the form `input like cell` (with a `concat()` call thrown in), where most of the times, queries are the other way around `cell like input`

I wasn't sure if sequelize could even do 2), but even if it could, I reckon the sequelize syntax would make the query very hard to read the query (likely mixing object and literal syntax in one statement ?)

In addition to that, this query does not return entities as instance of models or fetches dependencies, we just want 1 record of one email, if any theinput matches any, so the ORM brings no additional value here (minus maybe one level of indirection for the table name and field name).

There's also a bit of personal preference here too I suppose. Personally I think 
```
(expiry is NULL OR expiry > NOW())
```
is much easier to read than:
```
[Op.or]: [{ [Op.is]: null }, { [Op.gt]: new Date() }],
```
and that's just a small part of the statement in this case.

So I was keen to show that it's OK to have efficient readable raw queries in app code. ORM is just a tool, and shold not get in the way of good query patterns.

But so with all that said, If you can write the query in sequelize syntax, and it reads well, I don't mind pushing that in at all. It's your code base after all! 😄

Reply:
No worries Tim, agreed on the points you mentioned. I had some thoughts about mixing raw queries and ORMs, but understandability of code is more important than just have "standardised" code

Reply:
I've updated the comment to better explain why the raw query was used: https://github.com/isomerpages/isomercms-backend/pull/1301/commits/db15356463310f164287a44132d39c9eca2bd888